megamek/src/megamek/common/Entity.java

```diff
@@ -868,30 +875,19 @@ protected void initMilitary() {
     }
 
     protected boolean hasViableWeapons() {
-        int totalDmg = 0;
+        int totalDmg = Compute.computeTotalDamage(getTotalWeaponList());
```

Comment:
I feel like we could also optimize this for when we reach more than 5 total damage, but I guess that's in another loop now.

Reply:
I'd need to learn a bit more about Java concurrency; I attempted to put the incoming damage lists in each Entity but ended up with mismatched states and nothing worked so I pushed the list into Princess instead.

It would probably be a good idea to see how much stuff each Entity currently stores, how many times the various calculations get called, and whether that could all be sped up by caching... maybe a separate RFE?

Reply:
I started caching some things in the CachedEntityState class to optimize performance a little bit. I agree that it's beyond the scope of this PR.